File: Platforms/Bioma/BiomaSurfacePartonSoilSWATC/src/java/BiomaSurfacePartonSoilSWATC/SurfacePartonSoilSWATCComponent.java
public class SurfacePartonSoilSWATCComponent
{
    
    public SurfacePartonSoilSWATCComponent() { }

    SurfaceTemperatureParton _SurfaceTemperatureParton = new SurfaceTemperatureParton();
    SoilTemperatureSWAT _SoilTemperatureSWAT = new SoilTemperatureSWAT();

    public Double getLagCoefficient()
    { return _Soiltemperatureswat.getLagCoefficient(); }
    public void setLagCoefficient(Double LagCoefficient)
    { _Soiltemperatureswat.setLagCoefficient(LagCoefficient); } 
    public void  Calculate_surfacepartonsoilswatc(SurfacePartonSoilSWATCState s, SurfacePartonSoilSWATCState s1, SurfacePartonSoilSWATCRate r, SurfacePartonSoilSWATCAuxiliary a, SurfacePartonSoilSWATCExogenous ex)
    {
        _Surfacetemperatureparton.Calculate_surfacetemperatureparton(s, s1, r, a, ex);
        _Soiltemperatureswat.Calculate_soiltemperatureswat(s, s1, r, a, ex);
    }
    private Double LagCoefficient;
    public SurfacePartonSoilSWATCComponent(SurfacePartonSoilSWATCComponent toCopy) // copy constructor 
    {
        this.LagCoefficient = toCopy.getLagCoefficient();

    }
}

File: Platforms/Bioma/BiomaSurfacePartonSoilSWATC/src/cs/BiomaSurfacePartonSoilSWATC/SurfacePartonSoilSWATCComponent.cs
public class SurfacePartonSoilSWATCComponent
{
    
        public SurfacePartonSoilSWATCComponent() { }
    

    //Declaration of the associated strategies
    SurfaceTemperatureParton _SurfaceTemperatureParton = new SurfaceTemperatureParton();
    SoilTemperatureSWAT _SoilTemperatureSWAT = new SoilTemperatureSWAT();

    public double LagCoefficient
    {
        get
        {
             return _SoilTemperatureSWAT.LagCoefficient; 
        }
        set
        {
            _SoilTemperatureSWAT.LagCoefficient = value;
        }
    }

    public void  CalculateModel(SurfacePartonSoilSWATCState s, SurfacePartonSoilSWATCState s1, SurfacePartonSoilSWATCRate r, SurfacePartonSoilSWATCAuxiliary a, SurfacePartonSoilSWATCExogenous ex)
    {
        _surfacetemperatureparton.CalculateModel(s,s1, r, a, ex);
        _soiltemperatureswat.CalculateModel(s,s1, r, a, ex);
    }
    
    public SurfacePartonSoilSWATCComponent(SurfacePartonSoilSWATCComponent toCopy): this() // copy constructor 
    {

        LagCoefficient = toCopy.LagCoefficient;
    }
}